packages/react-menu/src/components/MenuTrigger/useTriggerElement.ts

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { useMergedRefs } from '@fluentui/react-utilities';
+import { useKeyboardNavigationState } from '@fluentui/react-focus-management';
+import { MenuTriggerState } from './MenuTrigger.types';
+import { useMenuContext } from '../../menuContext';
+
+// Helper type to select on parts of state the hook uses
+type UseTriggerElementState = Pick<MenuTriggerState, 'children'>;
+
+export const useTriggerElement = (state: UseTriggerElementState): MenuTriggerState => {
+  const triggerRef = useMenuContext(context => context.triggerRef);
+  const setOpen = useMenuContext(context => context.setOpen);
+  const open = useMenuContext(context => context.open);
+  const triggerId = useMenuContext(context => context.triggerId);
+  const on = useMenuContext(context => context.on);
+  const { isNavigatingWithKeyboard } = useKeyboardNavigationState();
+
+  const child = React.Children.only(state.children);
+
+  const triggerProps: Partial<React.HTMLAttributes<HTMLElement>> = {
+    'aria-haspopup': true,
+    'aria-expanded': open,
+    id: triggerId,
+    ...(child.props || {}),
+  };
+  /**
+   * Opens menu when focused ONLY with keyboard focus
+   */
+  if (on.includes('focus')) {
+    triggerProps.onFocus = (e: React.FocusEvent) => {
+      if (isNavigatingWithKeyboard?.()) {
```

Comment:
```suggestion
      if (isNavigatingWithKeyboard()) {
```

Can we make it non-optional, so we will not need additional checks?

Reply:
set a default that always returns false